api/index.js

```diff
@@ -1,89 +1,99 @@
+    ...customHeaders,
   }
+}
+
+const prepareRequest = init => {
+  const request = {}
+
+  request.url = prepareUrl(init.url, API_URL)
+  request.url.search = prepareParams({ ...init, url: request.url })
+  request.url = request.url.toString()
+  request.method = prepareMethod(init)
+  request.headers = prepareHeaders(init)
+
+  const body = prepareBody({ ...init, method: request.method })
+  if (body != null) request.body = body
+
+  request.credentials = 'include'
 
-  // Process body
-  if (!['GET', 'HEAD'].includes(method.toUpperCase())) {
-    if (typeof data == 'object') {
-      requestOptions.body = JSON.stringify(data)
-      requestOptions.headers['Content-Type'] = 'application/json'
-    } else requestOptions.body = data
+  return { ...init, ...request }
+}
+
+const processStatus = response => {
+  if (response.status >= 400) {
+    const Error = NetworkError.getErrorFromStatusCode(response.status)
+    throw new Error(
+      `Request for ${response.url} failed on ${response.status}`,
+      response
+    )
   }
 
-  // Process custom headers
-  Object.assign(requestOptions.headers, customHeaders)
-
-  // Make abortable
-  const controller = new AbortController()
-  const { signal } = controller
-  requestOptions.signal = signal
-
-  const fetchPromise = fetch(requestUrl, requestOptions)
-    .then(response => {
-      const { headers } = response
-      const type = headers.get('Content-Type')
-      const result = { type, headers }
-
-      if (response.status >= 400) {
-        const Error = NetworkError.getErrorFromStatusCode(response.status)
-        throw new Error(
-          `Request for ${method} ${url} failed on ${response.status}`,
-          response
-        )
-      }
-
-      if (method.toUpperCase() === 'HEAD') return result
-
-      const dataPromise = /application\/([\w.-]\+)?json/g.test(type)
-        ? response.json()
-        : response.blob()
-
-      return new Promise((resolve, reject) =>
-        dataPromise.then(
-          value =>
-            resolve({
-              data: value,
-              type,
-              headers,
-            }),
-          reason => reject(reason)
-        )
-      )
-    })
-    .catch(async error => {
-      const { response, message } = error
-      if (error instanceof NetworkError) throw error
-
-      // user is probably offline or CORS failed
-      throw new NetworkError(
-        `Request for ${method} ${url} failed. Response: ${response.status}, ${message}`
-      )
-    })
-
-  return new CancelablePromise(fetchPromise, {
-    cancel: () => controller.abort(),
-  })
+  return response
+}
+
+const processBody = response => {
+  const { headers } = response
+  const type = headers.get('Content-Type')
+
+  return (/application\/([\w.-]\+)?json/g.test(type)
+    ? response.json()
+    : response.blob()
+  ).then(data => ({ data, type, headers }))
+}
+
+// TODO: Remove later
+// Old type of error processing
+// eslint-disable-next-line no-unused-vars
```

Comment:
Why didn't you remove it?

Reply:
It would be good to remove it but I decided to do it in the next refactoring step. It's only my personal weakness to delete what just created. Sorry